### PR TITLE
feat: wait for Core to be synced before starting

### PR DESCRIPTION
**Pull request by jawid-h**

### CODE REVIEW

File: lib/core/checkCoreSyncFinishedFactory.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    const {
+      result: {
+        IsBlockchainSynced: isBlockchainSynced,
+      },
+    } = await coreRpcClient.mnsync('status');
+
+    if (!isBlockchainSynced) {
+      await wait(10000);
+
+      const result = await checkCoreSyncFinished(progressCallback);
```
Comment (shuplenkov) [inline lib/core/checkCoreSyncFinishedFactory.js:39]: return await?

Comment (shuplenkov) [inline lib/core/checkCoreSyncFinishedFactory.js:39]: Recursive async functions is not a good idea. It could cause stack overflow.

Comment (shumkov) [reply lib/core/checkCoreSyncFinishedFactory.js:39]: Make sense!

File: lib/core/checkCoreSyncFinishedFactory.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+   *
+   * @typedef checkCoreSyncFinished
+   *
+   * @param {function(number, number)} progressCallback
+   *
+   * @returns {Promise<Boolean>}
+   */
+  async function checkCoreSyncFinished(progressCallback) {
+    while (true) {
+      const {
```
Comment (shumkov) [inline lib/core/checkCoreSyncFinishedFactory.js:24]: I would move this call under `isBlockchainSynced` check 'coz if it's sync then there is no progress :)

File: scripts/abci.js [hunk 3]
```diff
@@ -9,6 +9,16 @@ const errorHandler = require('../lib/errorHandler');
 (async function main() {
   const container = await createDIContainer(process.env);
 
+  const logger = container.resolve('logger');
+
+  logger.info('checking Core has finished syncing...');
```
Comment (shumkov) [inline scripts/abci.js:14]: It could look strange in case if Core is synced already. Let's just remove it.

File: scripts/abci.js [hunk 4]
```diff
@@ -9,6 +9,16 @@ const errorHandler = require('../lib/errorHandler');
...
+  const checkCoreSyncFinished = container.resolve('checkCoreSyncFinished');
+  await checkCoreSyncFinished((currentBlockHeight, currentHeaderNumber) => {
+    logger.info(
+      `currently synced ${currentBlockHeight} blocks and ${currentHeaderNumber} headers`,
```
Comment (shumkov) [inline scripts/abci.js:18]: wait for Core is syncing ${currentBlockHeight}/${currentHeaderNumber}

File: scripts/abci.js (hunk 3 above)
Comment (jawid-h) [reply scripts/abci.js:14]: @shumkov I believe it's handy to see it in the logs to understand where are we now. Maybe it is better to leave it?

**Pull request merged**